### PR TITLE
build(deps): add dependabot configuration

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [x] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**
- [x] Enables [dependabot](https://dependabot.com/) to ensure all dependencies are always up to date!

PS. I've set the default reviewer to me, but it can be anyone from the core team.
// cc @bencodezen, @ulivz

### CODE REVIEW

**Comment by meteorlxy** [comment]
I think this should be categorized as "build-related changes" rather than a feature

**Comment by vinayakkulkarni** [comment]
@dependabot badge me

**Comment by bencodezen** [comment]
@vinayakkulkarni My apologies if this sounds like a stupid question. So does this PR help to enforce package.json / lock files in people's PRs going forward? I'm not familiar with the changes your submitting, so appreciate any guidance you can provide!

**Comment by vinayakkulkarni** [comment]
Sorry about the late response,

The config does not enforce anything in peoples' PRs, it basically removes user intervention in bumping the package dependencies version.

Basically, automates #2157 via dependabot Bot so that the package dependencies are automatically up-to-date.

**Comment by newsbielt703** [comment]
Agree that we need a bot to keep updating dependencies. However, I don't have many experience in it. What the pros and cons between Renovate and Dependabot?

**Comment by vinayakkulkarni** [comment]
> Agree that we need a bot to keep updating dependencies. However, I don't have many experience in it. What the pros and cons between Renovate and Dependabot?

I've been using dependabot for a long time now and now since it's owned by GitHub, it's even more awesomer!

Last time I tried using renovate, i had some billing related issue.. will have to explore Renovate pricing & plans though.

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+update_configs:
+  # Keep package.json (& lockfiles) up to date as soon as
+  # new versions are published to the npm registry
+  - package_manager: "javascript"
+    directory: "/"
+    update_schedule: "live"
+    # Apply default reviewer and label to created
+    # pull requests
+    default_reviewers:
+      - "vinayakkulkarni"
```
Comment (bencodezen) [inline .dependabot/config.yml:11]:
```suggestion
      - "bencodezen"
```

**Comment by bencodezen** [comment]
Thanks for working on this @vinayakkulkarni!

**Comment by vinayakkulkarni** [comment]
> Thanks for working on this @vinayakkulkarni!

<3

btw, dependabot was acquired by GitHub, I'll create a new PR to support v2 of dependabot

**Pull request merged**